api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -115,4 +115,7 @@ message HeaderValidatorConfig {
   // Envoy will respond with 400 to requests with disallowed methods.
   // By default methods with arbitrary names are accepted.
   bool restrict_http_methods = 3;
+
+  // Determines whether headers that contain underscores in their names should be accepted or not
```

Comment:
nit: end with '.'.

Just to make sure, is the default behavior of accepting underscores (`reject_headers_with_underscores = false`) the intended behavior?

Reply:
Yeah, the default behavior is to allow underscores. From my understanding we copied the default behavior of the [`headers_with_underscores_action`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-field-config-core-v3-httpprotocoloptions-headers-with-underscores-action) configuration setting.